test/test_testing.py

```diff
@@ -740,36 +747,103 @@ def test_trivial_passing_test(self, device):
         self.assertNotIn('OK', stderr.decode('ascii'))
 
 
+T = TypeVar("T", torch.Tensor, Sequence[torch.Tensor], Mapping[Any, torch.Tensor])
+
+
 class TestAsserts(TestCase):
-    def assert_fns(self):
-        return [torch.testing.assert_tensors_equal, torch.testing.assert_tensors_close]
+    def get_assert_fns(self) -> List[Callable]:
+        """Gets assert functions to be tested.
+
+        Returns:
+            List(Callable): Top-level assert functions from :mod:`torch.testing`.
+        """
+        return [torch.testing.assert_equal, torch.testing.assert_close]
+
+    def make_inputs(self, actual: torch.Tensor, expected: torch.Tensor) -> List[Tuple[T, T]]:
+        """Makes inputs for assert functions based on two example tensors.
+
+        Args:
+            actual (torch.Tensor): Actual tensor.
+            expected (torch.Tensor): Expected tensor.
+
+        Returns:
+            List[Tuple[T, T]]: Pairs of tensors, tensor sequences (:class:`tuple`, :class:`list`), and tensor mappings
+                (:class:`dict`, :class:`~collections.OrderedDict`)
+        """
+        return [
+            (actual, expected),
+            ((actual,), (expected,)),
+            ([actual], [expected]),
+            ({"t": actual}, {"t": expected}),
+            (collections.OrderedDict([("t", actual)]), collections.OrderedDict([("t", expected)])),
+        ]
+
+    def assert_fns_with_inputs(self, actual: torch.Tensor, expected: torch.Tensor) -> Iterator[Callable]:
+        """Yields assert functions with with included positional inputs based on two example tensors.
+
+        .. note::
+
+            This is a valid product of combinations from :meth:`get_assert_fns` and :meth:`make_inputs`. Every test
+            that does not test for anything specific should iterate over this to maximize the coverage.
+
+        Args:
+            actual (torch.Tensor): Actual tensor.
+            expected (torch.Tensor): Expected tensor.
+
+        Yields:
+            List[Callable]: Assert functions with predefined positional inputs.
+        """
+        for assert_fn, inputs in itertools.product(self.get_assert_fns(), self.make_inputs(actual, expected)):
+            yield functools.partial(assert_fn, *inputs)
+
+    @contextlib.contextmanager
+    def assertRaisesRegexs(self, expected_exception: Type[Exception], *expected_regexs: str) -> Iterator[None]:
```

Comment:
Don't we already have this context manager in the test suite?

Reply:
I don't think so. At least I didn't find it. This is basically an upgrade to the builtin `assertRaisesRegex` (mind the missing "s" at the end) that accepts multiple regexs. If you want I can move it to our `TestCase`.

Without something like that it is really hard to check if an error message contains multiple unordered parts. For an example see `test_unknown_type` (L1013-L1020): We want to make sure that the type of `actual` and `expected` appear in the message, but it doesn't matter which comes first.